packages/dds/sequence/src/generateSequentialId.ts

```diff
@@ -0,0 +1,144 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const digitZero = "A";
+const digitLast = "Z";
+const digitBeforeZero = "@";
+const digitAfterLast = "[";
+const idBase = 26;
+/**
+ * DomainSize is the number of zeroes generated ('A'-s)  before the first significant digit ('Z')
+ * for the initial SEQID. The parameter can be anything from zero to a bigger number.
+ */
+const domainSize = 5;
+
+export function isValidSeqId(seqId: string): boolean {
+  for (let i = 1; i < seqId.length; i++) {
+    const letterCode = getLetterCode(seqId[i]);
+    if (letterCode < 0 && letterCode >= idBase) {
+      return false;
+    }
+  }
+  return true;
+}
+
+const nextChar = (char: string) => String.fromCharCode(char.charCodeAt(0) + 1);
+
+const previousChar = (char: string) => String.fromCharCode(char.charCodeAt(0) - 1);
+
+// Returns value for UpperCase letter from 0 - 25;
+const getLetterCode = (char: string) => char.charCodeAt(0) - digitZero.charCodeAt(0);
+
+export function getDistance(seqId1: string, seqId2: string): number | undefined {
+  let numberSeqId1 = 0;
+  let numberSeqId2 = 0;
+
+  if (!isValidSeqId(seqId1)) {
+    throw new Error("seqId1 is not valid");
+  }
+
+  if (!isValidSeqId(seqId2)) {
+    throw new Error("seqId2 is not valid");
+  }
+
+  for (let i = 1; i < seqId1.length; i++) {
+    numberSeqId1 += getLetterCode(seqId1[i]) * (1 / Math.pow(idBase, i));
+  }
+
+  for (let i = 1; i < seqId2.length; i++) {
+    numberSeqId2 += getLetterCode(seqId2[i]) * (1 / Math.pow(idBase, i));
+  }
+  return numberSeqId1 - numberSeqId2;
```

Comment:
Remember, we are building infinite precision number? :)

You will likely hit pretty quickly precision limits here. It's probably fine for prototype, but I think for shipping code we would need to re-evaluate all the math and likely use custom types here (i.e. not number).
I do not see any usage of this function (is it a bug?), so hard to say what would be best representation of distance. It's likely same sequential id (i.e. we need to define math on sequential IDs).

Reply:
Resolving as this is oudated.